src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/__tests__/TargetingRulesTab.test.tsx

```diff
@@ -48,309 +48,491 @@ describe('TargetingRulesTab', () => {
+    test('it should use default onVariation if environment variation does not exist', async () => {
+      renderComponent({
+        featureFlagData: {
+          ...mockFeature,
+          envProperties: undefined
+        }
+      })
 
-  test('it should render target groups tags/dropdown correctly', async () => {
-    renderComponent()
-
-    // assert correct target groups are prepopulated
-    const targetGroupTagInputValue = screen.getByTestId('false_target_groups').querySelectorAll('span[data-tag-index]')
-
-    expect(targetGroupTagInputValue[0]).toHaveTextContent('target_group_4')
-    expect(targetGroupTagInputValue[1]).toHaveTextContent('target_group_5')
-
-    // assert target groups list appear on click
-    const targetGroupTagInput = screen.getByTestId('false_target_groups').querySelector('input') as HTMLInputElement
-    userEvent.type(targetGroupTagInput, 'target')
-    const targetGroupInputList = document.querySelector('ul')
-    await waitFor(() => expect(targetGroupInputList).toBeInTheDocument())
-    expect(document.querySelectorAll('li')[0]).toHaveTextContent(/target_group_5/)
-    expect(document.querySelectorAll('li')[1]).toHaveTextContent(/target_group_4/)
-    expect(document.querySelectorAll('li')[2]).toHaveTextContent(/target_group_1/)
-    expect(document.querySelectorAll('li')[3]).toHaveTextContent(/target_group_2/)
-    expect(document.querySelectorAll('li')[4]).toHaveTextContent(/target_group_3/)
+      expect(document.querySelector('input[name="onVariation"]') as HTMLSelectElement).toHaveValue('True')
+    })
   })
 
-  test('it should render target tags/dropdown correctly', async () => {
-    renderComponent()
+  describe('Variation Target/Target Groups', () => {
+    test('it should render target groups tags/dropdown correctly', async () => {
+      renderComponent()
+
+      // assert correct target groups are prepopulated
+      const targetGroupTagInputValue = screen
+        .getByTestId('false_target_groups')
+        .querySelectorAll('span[data-tag-index]')
+
+      expect(targetGroupTagInputValue[0]).toHaveTextContent('target_group_4')
+      expect(targetGroupTagInputValue[1]).toHaveTextContent('target_group_5')
+
+      // assert target groups list appear on click
+      const targetGroupTagInput = screen.getByTestId('false_target_groups').querySelector('input') as HTMLInputElement
+      userEvent.type(targetGroupTagInput, 'target')
+      const targetGroupInputList = document.querySelector('ul')
+      await waitFor(() => expect(targetGroupInputList).toBeInTheDocument())
+      expect(document.querySelectorAll('li')[0]).toHaveTextContent(/target_group_5/)
+      expect(document.querySelectorAll('li')[1]).toHaveTextContent(/target_group_4/)
+      expect(document.querySelectorAll('li')[2]).toHaveTextContent(/target_group_1/)
+      expect(document.querySelectorAll('li')[3]).toHaveTextContent(/target_group_2/)
+      expect(document.querySelectorAll('li')[4]).toHaveTextContent(/target_group_3/)
```

Comment:
Again, I know it's tidied up code, but this is a bit wasteful.
```suggestion
      
      const listItems = screen.getAllByRole('listitem')
      
      expect(listItems[0]).toHaveTextContent(/target_group_5/)
      expect(listItems[1]).toHaveTextContent(/target_group_4/)
      expect(listItems[2]).toHaveTextContent(/target_group_1/)
      expect(listItems[3]).toHaveTextContent(/target_group_2/)
      expect(listItems[4]).toHaveTextContent(/target_group_3/)
```